miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -113,11 +153,56 @@ internal class RealMiniApp(
         }
     }
 
+    private fun isRequiredPermissionDenied(appId: String): Boolean {
+        getCachedRequiredPermissions(appId).find {
+            it.second != MiniAppCustomPermissionResult.ALLOWED
+        }?.let { return true }
+        return false
+    }
+
+    private fun getCachedRequiredPermissions(
```

Comment:
Could you move `getCachedRequiredPermissions` and `getCachedOptionalPermissions` into `MiniAppManifestCache`? It seems like it would be better encapsulated in that class.